designs/2019-changing-base-path-in-config-files-that-cli-options-specify/README.md

```diff
@@ -0,0 +1,94 @@
+
+Git doesn't have `--ignore-path`-like CLI option, but has `core.excludesFile` setting to specify additional ignore file.
+
+```bash
+# Use './config/ignore' file as '.gitignore'.
+git config core.excludesFile config/ignore
+```
+
+In this case, the base path of the paths in `config/ignore` is the repository root. For example, `/node_modules` in `config/ignore` is `./node_modules` rather than `./config/ignore/node_modules`. This behavior is different from our `--ignore-path` option.
+
+## Detailed Design
+
+It changes the base path of the following patterns to the current working directory:
+
+- In the file which is specified by the `--config` option:
+    - `overrides[i].files`
+    - `overrides[i].excludedFiles`
+    - `ignorePatterns`
+- In the file which is specified by the `--ignore-path` option:
+    - file patterns.
+
+```bash
+# the paths of 'overrides' and 'ignorePatterns' are relative to './'
+eslint lib --config node_modules/@me/my-config/.eslintrc.js
+# the paths in `.eslintignore` are relative to './'
+eslint lib --ignore-path node_modules/@me/my-config/.eslintignore
+```
+
+This RFC *doesn't* change the resolving logic of `extends`, `parser`, and `plugins`.
+
+### Implementation
+
+It modifies [`createCLIConfigArray()` function](https://github.com/eslint/eslint/blob/869f96aa87c4f990f54e1eeccb0e3f7dbd66e6c2/lib/cli-engine/cascading-config-array-factory.js#L133-L165). In the function, it modifies the elements of the config arrays which are created from the `ignorePath` (corresponds to `--ignore-path`) and the `specificConfigPath` (corresponds to `--config`). It modifies `element.criteria.basePath` (corresponds to `overrides`) and `element.ignorePattern.basePath` (corresponds to `ignorePatterns` and `.eslintignore`) to `cwd`.
+
+## Documentation
+
+This change needs the migration guide because of a breaking change.
+
+This change needs to update the following documents:
+
+- the "[Configuration Based on Glob Patterns » Relative glob patterns](https://github.com/eslint/eslint/blob/869f96aa87c4f990f54e1eeccb0e3f7dbd66e6c2/docs/user-guide/configuring.md#relative-glob-patterns)" section.
+- the "[Ignoring Files and Directories » `ignorePatterns` in config files](https://github.com/eslint/eslint/blob/869f96aa87c4f990f54e1eeccb0e3f7dbd66e6c2/docs/user-guide/configuring.md#ignorepatterns-in-config-files)" section.
+- the "[Ignoring Files and Directories » `.eslintignore`](https://github.com/eslint/eslint/blob/869f96aa87c4f990f54e1eeccb0e3f7dbd66e6c2/docs/user-guide/configuring.md#eslintignore)" section.
+    - As a side note, this section is outdated a bit. It said "Paths are relative to `.eslintignore` location or the current working directory.", but it has been changed since ESLint 4.0.0. Currently, it's "Paths are relative to `.eslintignore` location."
+
+## Drawbacks
+
+This is a breaking change. It can break the current workflow.
```

Comment:
I see why it can be considered a breaking change but I'm sure there are also some folks which would be fine with calling it a bug fix as well, just saying 😄

Reply:
This is definitely a breaking change. The current behavior was produced in 4.0.0 since users requested. This RFC changes the requested behavior.